easybuild/tools/filetools.py

```diff
@@ -2668,17 +2667,34 @@ def get_source_tarball_from_git(filename, targetdir, git_config):
             for cmd in cmds:
                 run_shell_cmd(cmd, work_dir=work_dir, hidden=True, verbose_dry_run=True)
 
-    # create an archive and delete the git repo directory
+    # Create archive
+    archive_path = os.path.join(target_dir, filename)
+
     if keep_git_dir:
-        tar_cmd = ['tar', 'cfvz', targetpath, repo_name]
+        # create archive of git repo including .git directory
+        tar_cmd = ['tar', 'cfvz', archive_path, repo_name]
     else:
-        tar_cmd = ['tar', 'cfvz', targetpath, '--exclude', '.git', repo_name]
+        # create reproducible archive
+        # see https://reproducible-builds.org/docs/archives/
+        tar_cmd = [
+            # print names of all files and folders excluding .git directory
+            'find', repo_name, '-name ".git"', '-prune', '-o', '-print0',
+            # reset access and modification timestamps
+            '-exec', 'touch', '-t 197001010100', '{}', '\;', '|',
```

Comment:
I missed using a raw string for this bit:
```suggestion
            '-exec', 'touch', '-t 197001010100', '{}', r'\;', '|',
```